packages/ListBox/src/components/Option/Option.js

```diff
@@ -2,12 +2,55 @@ import React from "react";
 import PropTypes from "prop-types";
 import useListBox from "../../useListBox";
 import { OnChangeContext } from "../../store/OnChangeProvider";
+import { PropsContext } from "../../store/PropsProvider";
 import { getA11yAttributesForOption } from "../../helpers/DOMAttributes";
 import { isOptionVisible, isOptionSelected, handleClickOption } from "../Options/helpers/options";
 import useIsSelectedOption from "./useIsSelectedOption";
 import * as sc from "./Option.styles";
 
-const propTypes = {
+export default function Option(props) {
+  const { index, groupId, label, ...moreProps } = props; // eslint-disable-line
+  const [state, dispatch] = useListBox();
+  const { activeOption, listBoxHasFocus } = state;
+  const providedProps = React.useContext(PropsContext);
+  const { isReadOnly, size } = providedProps;
+  const onChangeContext = React.useContext(OnChangeContext);
+
+  useIsSelectedOption({ index, props });
+
+  if (typeof state.options[index] === "undefined" || !isOptionVisible(state, index)) {
+    return null;
+  }
+
+  const isSelected = isOptionSelected(state, index);
+  const isDisabled = providedProps.isDisabled || props.isDisabled || isReadOnly;
```

Comment:
Mostly the code for this component was just moved around, very little changed. This is one change – items are disabled if `isReadOnly`.  This only applies when `isInline` since the dropdown list does not appear for a disabled popover-style component.